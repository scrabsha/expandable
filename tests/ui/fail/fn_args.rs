#[allow(unused_macros)]
#[expandable::item]
macro_rules! test {
    () => {
        fn test(,) -> u8 {
            42
        }
    };
}

#[allow(unused_macros)]
#[expandable::item]
macro_rules! test {
    () => {
        fn test(a: u8,,) -> u8 {
            42
        }
    };
}

#[allow(unused_macros)]
#[expandable::item]
macro_rules! test {
    () => {
        fn test(a) -> u8 {
            42
        }
    };
}

fn main() {}
