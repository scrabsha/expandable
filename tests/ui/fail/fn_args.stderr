error: Potentially invalid expansion. Expected a `)`, an identifier.
 --> tests/ui/fail/fn_args.rs:5:17
  |
5 |         fn test(,) -> u8 {
  |                 ^

error: Potentially invalid expansion. Expected a `)`, an identifier.
  --> tests/ui/fail/fn_args.rs:15:23
   |
15 |         fn test(a: u8,,) -> u8 {
   |                       ^

error: Potentially invalid expansion. Expected `:`.
  --> tests/ui/fail/fn_args.rs:25:16
   |
25 |         fn test(a) -> u8 {
   |                ^^^
